initialise/entrypoints/index.php

```diff
@@ -11,14 +11,17 @@
 require_once '/srv/mediawiki/config/initialise/MirahezeFunctions.php';
 require MirahezeFunctions::getMediaWiki( 'includes/WebStart.php' );
 
-if ( $wgArticlePath === '/$1' && str_contains( strtoupper( $_SERVER['REQUEST_URI'] ), strtoupper( '/wiki/' ) ) ) {
-	// Redirect to the same page maintaining the path
-	header( 'Location: ' . str_replace( '/wiki/', '/', $_SERVER['REQUEST_URI'] ), true, 301 );
-	exit();
-} elseif ( $wgArticlePath === '/wiki/$1' && !str_contains( $_SERVER['REQUEST_URI'], '/wiki/' ) && !str_contains( $_SERVER['REQUEST_URI'], '/w/' ) && !( $wgMainPageIsDomainRoot && $_SERVER['REQUEST_URI'] === '/' ) ) {
-	// Redirect to the same page maintaining the path
-	header( 'Location: /wiki' . $_SERVER['REQUEST_URI'], true, 301 );
-	exit();
+// T12214: Avoid redirects if action=raw is set
+if ( RequestContext::getMain()->getActionName() !== "raw" ) {
```

Comment:
A bit minor but for consistency we should use single quotes for raw

Reply:
Oops, it's kinda instinct for me to use double quotes since I've used other programming languages more than PHP 😅